### ISSUE TITLE
Feature: Data Driven Testing

### ISSUE THREAD

**Issue by suic86** [Codewars/codewars.com#664]
## Feature
Add libraries for data driven testing e. g. [ddt](http://ddt.readthedocs.io/en/latest/index.html) in Python.

## Rationale
1. More diversified testing environment supporting multiple types/styles of testing.
2. IMO: Simplifying kata craeation and translation: One can create/convert a data structure without writing/copying a lot of boilerplate code.

**Comment by jhoffner** [comment]
I'm not 100% sure yet what this is yet, but this may be a feature for the [CLI](github.com/Codewars/codewars-runner-cli)

**Comment by the-zebulan** [comment]
@suic86 ,

That seems like a neat library, thanks for the link.

I've been playing around with [Hypothesis](https://github.com/HypothesisWorks/hypothesis-python) on a few personal projects. I keep thinking how it could be useful for Codewars Python tests too. Adding a decorator to a `test.assert_equals` for random tests might help one get rid of one of the classic Codewars beta issues... `"needs random tests"`.

**Comment by suic86** [comment]
@jhoffner 

You're right it's for cli :) Can I migrate this to the CLI issue list? If not, I'm going to close this and re-raise the feature request there.

@the-zebulan

Nice one! This reminds me of [QuickCheck](https://hackage.haskell.org/package/QuickCheck) in Haskell which is supported on Codewars. However, I've had hard times with it. :) I agree that something like _hypotheses_ could help with random tests. There would be nice to have some random test functionality in the standard codewars test framework as random tests involve too much boilerplate code.

**Comment by the-zebulan** [comment]
@suic86 ,

You are absolutely right, I believe it is based on Haskell's QuickCheck.

> ***This is often called property based testing, and was popularised by the Haskell library Quickcheck.***

 I have only recently started to use it on some small projects so you probably have much more experience with property based testing. I'll have to try using `ddt` soon to compare the two for my own understanding.
